perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -682,6 +687,29 @@ private UserExtSource getUserExtSourceByUniqueAttributeValue(PerunSession sess,
 		return usersManagerImpl.getUserExtSourceByUniqueAttributeValue(sess, attrDef.getId(), uniqueValue);
 	}
 
+	@Override
+	public UserExtSource getUserExtSourceFromMultipleIdentifiers(PerunSession sess, PerunPrincipal principal) throws InternalErrorException, UserExtSourceNotExistsException {
+		String additionalIdentifiers = principal.getAdditionalInformations().get(additionalIdentifiersAttributeName);
+		if (additionalIdentifiers == null) {
+			throw new UserExtSourceNotExistsException("Mandatory attribute is not defined: ".concat(additionalIdentifiersAttributeName));
+		}
+		UserExtSource ues = null;
+		for(String identifier : additionalIdentifiers.split(multivalueAttributeSeparatorRegExp)) {
+			try {
+				ues = perunBl.getUsersManagerBl().getUserExtSourceByUniqueAttributeValue(sess, additionalIdentifiersPerunAttributeName, identifier);
+				log.debug("UserExtSource found using additional identifiers: " + ues);
+				break;
+			} catch (UserExtSourceNotExistsException ex) {
+				//try to find user ext source using different identifier in the next iteration of for cycle
+			} catch (InternalErrorException | AttributeNotExistsException ex) {
+				String errorMessage = "Mandatory attribute is not defined: ".concat(additionalIdentifiersPerunAttributeName);
+				log.error(errorMessage);
```

Comment:
There is really no need for creating a new string here. You can use the code immediately in the log.error() call.

Reply:
In that case, how will I throw the exception with the message on the next line? There was a requirement to log the error and then throw it if I remember well..